mmrazor/models/mutators/channel_mutator/dmcp_channel_mutator.py

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Any, Dict, Type, Union
+import random
+
+import torch
+import torch.nn as nn
+from torch.nn import Module, ModuleList
+from mmrazor.models.mutables import DMCPChannelUnit
+from mmrazor.registry import MODELS
+from .channel_mutator import ChannelMutator, ChannelUnitType
+from ...architectures import DMCPBatchNorm2d
+from ...task_modules.estimators import ResourceEstimator
+
+@MODELS.register_module()
+class DMCPChannelMutator(ChannelMutator[DMCPChannelUnit]):
+    """DMCP channel mutable based channel mutator. It uses DMCPPChannelUnit.
+
+    Args:
+        channel_unit_cfg (Union[dict, Type[ChannelUnitType]], optional):
+            Config of MutableChannelUnits. Defaults to
+            dict( type='DMCPPChannelUnit', units={}).
+        parse_cfg (Dict): The config of the tracer to parse the model.
+            Defaults to dict( type='BackwardTracer',
+                loss_calculator=dict(type='ImageClassifierPseudoLoss')).
+            Change loss_calculator according to task and backbone.
+        pruning_cfg (Tuple): (min_sample_rate, max_sample_rate, sample_offset)).
+    """
+
+    def __init__(self,
+                 channel_unit_cfg: Union[dict, Type[ChannelUnitType]] = dict(
+                     type='DMCPChannelUnit', units={}),
+                 parse_cfg=dict(
```

Comment:
update parse_cfg to `ChannelAnalyzer`, according to: https://github.com/open-mmlab/mmrazor/blob/26da4d223a1c215388b1a911317e5921cc288917/mmrazor/models/mutators/channel_mutator/channel_mutator.py#L73